prometheus/http.go

```diff
@@ -15,14 +15,169 @@ package prometheus
 
 import (
 	"bufio"
+	"compress/gzip"
+	"fmt"
 	"io"
+	"log"
 	"net"
 	"net/http"
 	"strconv"
 	"strings"
 	"time"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+const (
+	contentTypeHeader     = "Content-Type"
+	contentLengthHeader   = "Content-Length"
+	contentEncodingHeader = "Content-Encoding"
+	acceptEncodingHeader  = "Accept-Encoding"
+)
+
+// Handler returns an HTTP handler for the DefaultRegistry. It is
+// already instrumented with InstrumentHandler (using "prometheus" as handler
+// name).
+//
+// Please note the issues described in the doc comment of InstrumentHandler. You
+// might want to consider using UninstrumentedHandler instead.
+//
+// The returned Handler is using the same HandlerOpts as the Handler returned by
+// UninstrumentedHandler. See there for details.
+func Handler() http.Handler {
```

Comment:
On 2 August 2016 at 14:50, Brian Brazil <notifications@github.com> wrote:
>
> We can reduce the number of top-level identifiers by putting things in classes.

That's a purely formal argument. Sticking functions to structs is
sugarcoating and not reducing the mental load. But I said that again.

> The reason to split things out to other packages is to avoid dependencies in the core instrumentation, not to reduce top-level identifiers.

Packages are the Go way of grouping related functions.

> I'd overall expect that the Go client library would have roughly the same user-facing API as the Java client. The only top-level functions I'd expect in Go would be factories (Java has only one top-level function, for reasons that don't apply to Go).

I don't think it's a good idea to coerce the Go API into that of a
very different language.

-- 
Björn Rabenstein, Engineer
http://soundcloud.com/brabenstein

SoundCloud Ltd. | Rheinsberger Str. 76/77, 10115 Berlin, Germany
Managing Director: Alexander Ljung | Incorporated in England & Wales
with Company No. 6343600 | Local Branch Office | AG Charlottenburg  |
HRB 110657B